processing/src/main/java/org/apache/druid/collections/DefaultBlockingPool.java

```diff
@@ -91,26 +97,38 @@ public List<ReferenceCountingResourceHolder<T>> takeBatch(final int elementNum,
     Preconditions.checkArgument(timeoutMs >= 0, "timeoutMs must be a non-negative value, but was [%s]", timeoutMs);
     checkInitialized();
     try {
+      pendingQueries.incrementAndGet();
       final List<T> objects = timeoutMs > 0 ? pollObjects(elementNum, timeoutMs) : pollObjects(elementNum);
       return objects.stream().map(this::wrapObject).collect(Collectors.toList());
     }
     catch (InterruptedException e) {
       throw new RuntimeException(e);
     }
+    finally {
+      pendingQueries.decrementAndGet();
+    }
   }
 
   @Override
   public List<ReferenceCountingResourceHolder<T>> takeBatch(final int elementNum)
   {
     checkInitialized();
     try {
+      pendingQueries.incrementAndGet();
       return takeObjects(elementNum).stream().map(this::wrapObject).collect(Collectors.toList());
     }
     catch (InterruptedException e) {
+      pendingQueries.incrementAndGet();
```

Comment:
Seems like this is missing a finally block to decrement pendingQueries
```suggestion
```

Reply:
updated. Good catch